chain-core/src/common/merkle_tree.rs

```diff
@@ -142,24 +142,83 @@ impl<T> Tree<T> {
 ///
 /// leaf_hash is computed from the value in the Proof
 
-#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode)]
+#[derive(Debug, PartialEq, Eq, Clone)]
 pub enum Side {
     Left,
     Right,
 }
 
-#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode)]
+#[derive(Debug, PartialEq, Eq, Clone)]
 pub struct PathNode {
     node_hash: H256,
     child_hash: H256, // hash of left child unless reversed
     hash_side: Side,  // the side of hash child
 }
 
-#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode)]
+impl Encode for PathNode {
+    fn encode_to<EncOut: Output>(&self, dest: &mut EncOut) {
+        dest.push(&self.node_hash);
```

Comment:
When we want to add extra field in the future, what can we do? It seems no extra metadata for future extension.

Reply:
I don't foresee a need to add extra fields to Merkle proofs, as they are a part of witness payloads where all its data is minimal and critical (needs to be validated).
For transaction payload evolution, it'll be more meaningful to extend witness types (and corresponding output types or whatever will make sense in that scenario).

Reply:
transaction input witness can facilitate that https://github.com/crypto-com/chain/blob/1dd1ad38e6c3022666209630776eff6c43cd63a9/chain-core/src/tx/witness/mod.rs#L113